snapcraft/internal/errors.py

```diff
@@ -23,22 +23,80 @@
 class SnapcraftError(Exception):
     """Base class for all snapcraft exceptions.
 
-    :cvar fmt: A format string that daughter classes override
-
+    :cvar brief: Concise, single-line description of the error.
+    :cvar details: Detailed information, if considered necessary for user.
+    :cvar resolution: Concise suggestion for user to resolve error.
+    :cvar docs_url: docs.snapcraft.io link, if applicable.
+    :cvar exit_code: Exit code to use if this exception causes Snapcraft to exit.
+    :cvar reportable: Prompt user to report this error to sentry.
+
+    Format string options may be used for templating keyworded exceptions.
+
+    :cvar fmt: (DEPRECATED) Legacy format string.
+    :cvar fmt_brief: Keyworded format string for error.
+    :cvar fmt_details: Keyworded format string for details.
+    :cvar fmt_resolution: Keyworded format string for resolution.
     """
 
-    fmt = "Daughter classes should redefine this"
+    fmt: str = ""
+    fmt_brief: str = ""
+    fmt_resolution: str = ""
+    fmt_details: str = ""
+    docs_url: str = ""
+    exit_code: int = 2
+    reportable: bool = False
 
     def __init__(self, **kwargs) -> None:
+        self._kwargs = kwargs
+
+        # DEPRECATED: Please define real properties when creating classes.
```

Comment:
Maybe just create a new error class and make it an AbstractBaseClass while at it that we can start inheriting from and keep the old class as is.